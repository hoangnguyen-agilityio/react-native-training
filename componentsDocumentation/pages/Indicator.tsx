import React from 'react';
import { ActivityIndicator, StyleSheet, SafeAreaView } from 'react-native';
import Section from '../components/Section';
import Title from '../components/Title';

const IndicatorPage = () => (
  <SafeAreaView style={styles.container}>
    <Section>
      <Title>Default:</Title>
      <ActivityIndicator />
    </Section>
    <Section>
      <Title>Large Indicator with custom color:</Title>
      <ActivityIndicator size="large" color="#00ff00" />
    </Section>
    <Section>
      <Title>Small Indicator with custom color:</Title>
      <ActivityIndicator size="small" color="#0000ff" />
    </Section>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 16,
  },
});

export default IndicatorPage;
