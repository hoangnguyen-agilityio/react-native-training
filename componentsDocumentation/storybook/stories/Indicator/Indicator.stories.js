import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { ActivityIndicator } from 'react-native';
import CenterView from '../CenterView';

storiesOf('Indicator', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => <ActivityIndicator />)
  .add('Large Indicator', () => (
    <ActivityIndicator size="large" color="#00ff00" />
  ))
  .add('Small Indicator with custom color', () => (
    <ActivityIndicator size="small" color="#0000ff" />
  ));
