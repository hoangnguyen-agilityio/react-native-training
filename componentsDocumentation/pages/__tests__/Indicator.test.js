import React from 'react';
import renderer from 'react-test-renderer';
import Indicator from '../Indicator';
import {ActivityIndicator} from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<Indicator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('there are 3 indicators in the page', () => {
  const tree = renderer.create(<Indicator />);
  const indicators = tree.root.findAllByType(ActivityIndicator);

  expect(indicators).toHaveLength(3);
});