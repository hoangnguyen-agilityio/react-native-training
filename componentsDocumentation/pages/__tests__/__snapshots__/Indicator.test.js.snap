// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "marginHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#737373",
        "borderBottomWidth": 2,
        "marginTop": 24,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 12,
        }
      }
    >
      Default:
    </Text>
    <ActivityIndicator />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#737373",
        "borderBottomWidth": 2,
        "marginTop": 24,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 12,
        }
      }
    >
      Large Indicator with custom color:
    </Text>
    <ActivityIndicator
      color="#00ff00"
      size="large"
    />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#737373",
        "borderBottomWidth": 2,
        "marginTop": 24,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 12,
        }
      }
    >
      Small Indicator with custom color:
    </Text>
    <ActivityIndicator
      color="#0000ff"
      size="small"
    />
  </View>
</RCTSafeAreaView>
`;
